### ISSUE TITLE
TaskHandle: fix wrong `sender is dropped while join handle is still active` warnings

### ISSUE THREAD

**Issue by problame** [neondatabase/neon#2885]
> Do we fix the right thing here? I mean that this TaskHandle logic needs to be improved, but if task drops event_sender and continues running without returning it looks like a problem too

Conceptually `TaskHandle` provides an abstraction for getting progress updates on a spawned task.
If the task chooses to not send progress updates (= drop the `events_sender`) it shouldn't be considered an error.
Or should it?
Anyways, even if the sender doesn't "consciously" drop it, the fact that we move it into the task closure means that we **always** drop it before the task exits & `join_handle.await` returns. It's an inherent race. It's just so late in the task's life that more often than not, the task exists first before the `.recv()` returns `Err(_sender_dropped)`.

_Originally posted by @problame in https://github.com/neondatabase/neon/issues/2862#issuecomment-1323309414_

**Comment by LizardWizzard** [comment]
>  It's an inherent race.

Agreed. Ideally we should not give ownership over receiver to the task so TaskHandle can drop receiver after task finishes.

> it shouldn't be considered an error.
Or should it?

I can hardly imagine when this the desired behavior. In this case task is alive but you no longer can receive progress updates.

Also worth noting that TaskHandle has only one usage across the codebase, so as usual it is hard to come up with good generic solution when you have only one use case

**Comment by koivunej** [comment]
This most def is not a good first issue.
